### ISSUE TITLE
Question: Should an Incomplete/Resumed Job leave Unreferenced Chunks?

### ISSUE THREAD

**Issue by jonreeves** [gilbertchen/duplicacy#352]
The first thing to mention, is that my question pertains to a Source Folder whose content did not change at all during my Stop and Resume of a Backup Job.

**The Background**
The other day I started a Backup Job to a USB3.0 hard drive from a Source USB3.0 Hard Drive. I had to stop the backup a few times, while I was trying to figure out why the speed was slowing (#349). In the end this initial/1st Snapshot Backup job ran 4 Times (I stopped and restarted the job 3 Times).

I was initially concerned that each subsequent `.duplicacy/incomplete` file would eradicate the previous, but duplicacy appeared smart enough to add the new progress to the old and keep it accumulating, Phew.

The backup job finished not long ago, and in the end settled around 12MB/s, which is not great for Local Storage, but I'm convinced this is just because of the many Small Chunks being written to a traditional HDD. I am looking at running it again with a much larger Chunk Size given some recent analysis of my average file size etc...

**The Expectation**
Thanks to the `.duplicacy/incomplete` file I was expecting to beable to skip over a large number of chunks and therefore speed up the job. What struck me as strange was that this didn't appear to happen, or maybe it did, its not clear...

By my forth and final attempt it was said this upon start:
```
Listing all chunks
Skipped 279093 files from previous incomplete backup
```
I only just noticed that it explicitly says `files` and not `chunks` so I'm guessing that is why now that the job has concluded I was able to search through the log to see that `48,198` **chunks** were skipped, and not `279,093`.

In the interest of checking I didn't have some unreferenced chunks, I ran the following command:
```
duplicacy -log prune -exhaustive -exclusive -all -dry-run
```
It reported back that there were infact `65,859` unreferenced chunks. What I curious about, is how this happened?

- This was a new/empty repository before the initial backup attempt ran.
- The drive's content didn't change at all inbetween attempts.

Without fully appreciating how the chunks are determined I assumed that two runs on the same drive would yield the same chunks. If that is supposed to be the case then I could maybe offer up a few factors in my actions that might have made this not work as intended...

1. After attempt 1 I moved to a different Laptop, so that I could leave it running while away from home. The drive Letter changed, I recall having to edit the `preferences` file.
2. After Attempt 2, I edited the `preferences` file again to have an extra destination for a `samba://....` version path so I could see if the speed was different. This Storage had a different Name.
3. No changes where made After attempt 3.

**Questions**
Is it common / likely that Unreferenced Chunks occur even if the Source Files don't change?
How can you reduce the chance of producing Unreferenced Chunks?
Is Pruning for Unreferenced Chunks the best way to clean up unnesscessary files?

**Comment by gilbertchen** [comment]
If there isn't a `.duplicacy/incomplete` file, then a resumed backup will start from the beginning and will not create any unreferenced file.  But the `.duplicacy/incomplete` file causes a resumed backup to start at the first file that had not been completely uploaded in the previous attempt, and since the chunk boundaries do not usually align with files, so starting from this file may create a few new chunks.

It is not clear to me why you're seeing this many unreferenced chunks.  I ran a test on my 34GB repository and there were only 2 unreferenced chunks from multiple interrupts.

Any chance you're seeing any `Deleting snapshot <id> at revision <revision>` messages at the beginning of the output?  If so, then those referenced chunks are results of deleting some backups.

**Comment by jonreeves** [comment]
Yeah I'm not sure what caused the large number of unreferenced chunks. I just replayed my steps with a slightly smaller set of data; cancelling the job and resuming from another machine, and then switching from going `local` to `samba`. By the end there was only 1 unreferenced chunk and a few `.tmp` files that were left lingering.

I'm gonna assume it was human error and that the repository wasn't empty when I started the process because I'm not sure what else would have caused it.

I'll reopen the case, if I discover something more concrete.
